plugins/remove_get_trackers.py

```diff
@@ -0,0 +1,22 @@
+"""
+Remove GET trackers from URLs in sent messages.
+"""
+from poezio.plugin import BasePlugin
+import re
+
+class Plugin(BasePlugin):
+    def init(self):
+        self.api.add_event_handler('muc_say', self.remove_get_trackers)
+        self.api.add_event_handler('conversation_say', self.remove_get_trackers)
+        self.api.add_event_handler('private_say', self.remove_get_trackers)
+
+    def remove_get_trackers(self, msg, tab):
+        # fbclid: used globally (Facebook)
+        # utm_*: used globally https://en.wikipedia.org/wiki/UTM_parameters
+        # ncid: DoubleClick (Google)
+        # ref_src, ref_url: twitter
+        # Others exist but are excluded because they are not common.
+        # See https://en.wikipedia.org/wiki/UTM_parameters
+        msg['body'] = re.sub('&?(fbclid|dclid|ncid|utm_source|utm_medium|utm_campaign|utm_term|utm_content|ref_src|ref_url)=[^ &#]*',
```

Comment:
- This is missing the ``;`` separator sometimes used for session IDs.
- The start of the regex should be more like ``[&?]`` and it should treat the case that the first character is in fact a questionmark specially by not removing the questionmark.

I also think that it should ensure that it is operating within the context of an HTTP URL. And even then, I’m sceptical, because it’ll do bad things to e.g. code pastes which need to stay verbatim.

In addition, this does not fix the URLs which may be included in XHTML-IM.

Reply:
>     * This is missing the `;` separator sometimes used for session IDs.

From what I understand, this is not standardized, and a vulnerability of the website. It's not what I'm targeting with this plugin

> 
>     * The start of the regex should be more like `[&?]` and it should treat the case that the first character is in fact a questionmark specially by not removing the questionmark.

Having '?&' in a URL is still valid, so even if the tracker is the first parameter of a multi-parameters URL, it's still producing a valid link

> I also think that it should ensure that it is operating within the context of an HTTP URL. And even then, I’m sceptical, because it’ll do bad things to e.g. code pastes which need to stay verbatim.

I pushed a new commit to take account of the context.
In case this breaks things in code paste, remember that this is a plugin, so it can be deactivated

> In addition, this does not fix the URLs which may be included in XHTML-IM.

I don't know how to do that as I'm not using this mode